### PR TITLE
Update chatbot UI and update hallucination check

### PR DESCRIPTION
**Pull request by nehcgs**

### CODE REVIEW

**Review by salmanap — APPROVED** [review]

**Comment by cotran2** [comment]
Including new fix on the hallucination check:
1. Change system-role -> assistant-role (potential confusion for gpt if see system role in history)
2. add metadata for the hallucination chat completion request

Need review from @adilhafeez

**Comment by adilhafeez** [comment]
@cotran2 why are we updating hallucination?

**Comment by cotran2** [comment]
We observe a bug that if the role of the chat template for hallucination check is system role, it will cause gpt to be confused (correct here is assistant).

File: crates/prompt_gateway/src/hallucination.rs [hunk 1]
```diff
@@ -18,9 +18,11 @@ pub fn extract_messages_for_hallucination(messages: &Vec<Message>) -> Vec<String
...
             if let Some(model) = message.model.as_ref() {
                 if !model.starts_with(ARCH_MODEL_PREFIX) {
-                    if message.role == ASSISTANT_ROLE {
-                        break;
-                    }
+                  if let Some(content) = &message.content {
+                      if !content.starts_with(HALLUCINATION_TEMPLATE) {
+                          break;
+                      }
+                  }
```
Comment (adilhafeez) [inline crates/prompt_gateway/src/hallucination.rs:25]: is this needed

Comment (cotran2) [reply crates/prompt_gateway/src/hallucination.rs:25]: Yes, this is for separating the hallucination template. When we collect the user history message, we want to go pass over the chat template if there is parameter gathering from ArchFC.

Comment (cotran2) [reply crates/prompt_gateway/src/hallucination.rs:25]: Yes, this is for separating the hallucination template. When we collect the user history message, we want to go pass over the chat template if there is parameter gathering from ArchFC.

Comment (adilhafeez) [reply crates/prompt_gateway/src/hallucination.rs:25]: can you add test case to simulate this behavior

Comment (cotran2) [reply crates/prompt_gateway/src/hallucination.rs:25]: the current test case reflects this.

**Pull request merged**